app/src/main/java/com/example/android/tvleanback/ui/PlaybackOverlayFragment.java

```diff
@@ -189,16 +201,15 @@ public void onActionClicked(Action action) {
                 if (action.getId() == mPlayPauseAction.getId()) {
                     togglePlayback(mPlayPauseAction.getIndex() == PlayPauseAction.PLAY);
                 } else if (action.getId() == mSkipNextAction.getId()) {
-                    next();
+                    next(mCurrentPlaybackState == PlaybackState.STATE_PLAYING);
                 } else if (action.getId() == mSkipPreviousAction.getId()) {
-                    prev();
+                    prev(mCurrentPlaybackState == PlaybackState.STATE_PLAYING);
                 } else if (action.getId() == mFastForwardAction.getId()) {
                     fastForward();
                 } else if (action.getId() == mRewindAction.getId()) {
                     fastRewind();
                 }
                 if (action instanceof PlaybackControlsRow.MultiAction) {
-                    ((PlaybackControlsRow.MultiAction) action).nextIndex();
```

Comment:
syncing the playback state with the UI apperance now is done in the MediaController.Callback. Here in the OnActionClickListener we only fire the command. The result of firing the command will be a change in the PlaybackState which then is notified back via the MediaController.Callback